app/Http/Controllers/Auth/OAuthController.php

```diff
@@ -80,9 +78,13 @@ public function handleProviderCallback(string $provider): RedirectResponse
             return redirect(route('frontend.profile.index'));
         }
 
-        $user = User::where($provider.'_id', $providerUser->getId())->first();
+        $user = User::where($provider.'_id', $providerUser->getId())->orWhere('email', $providerUser->getEmail())->first();
```

Comment:
There's a `discord_id` in the `User` table? I think this is maybe better to be in the auth tokens table

Reply:
It was already there before I even started working on oauth. Don't you think that having the id in the `User` table makes it simpler in terms of queries? Otherwise that means that you have to try to match the user either via their `email` in the `User` table or find a user via the id in the `UserOAuthToken` table and then from this row retrieve the user. This increases the number of queries.

Reply:
Yes @nabeelio , you added both `discord_id` and `discord_privatechannel_id` many months ago, maybe more than a year ago when you first implemented some of the discord notifications. Told people that more stuff will come soon and they will be used etc.

We never used those fields though, only implemented webhooks for discord notifications, nothing more. @arthurpar06 used the discord_id field properly with the oAuth implementation.

Maybe we can get user's private channel id's with oAuth and permissions too (I am not sure of course) but if this can happen, it will be great for v7 to be able to send direct messages to users. (like rejection of their pireps, comments about their pireps, status changes, awards etc)

Reply:
I can easily do that; I did it in one of the early PRs, but when I cleaned it up, I removed it because Nabeel didn't see much use for it. 

However, I can add it back; it would be cool. What would be nice if we do that (it would require some work) is to create a kind of notification settings page where each user can opt-in for a notification and choose the channel on which they want to receive it. Like a real notification center

Reply:
I do not think that we would need additional settings for administrative stuff. If a user registers and links his/her discord account, then this will be more than enough to send personal notifications.

Do we ask their permission for sending pirep related mails ? No :) Why do we ask permission for discord notifications if they link their accounts.

What can be sent;

* Pirep Related Notifications : Accepted, Rejected, Pending
* Account Related Notifications : On Leave, Suspended
* Profile Related Notifications : Rank Change, Award Issued
* Addon Related Notifications (if developed by third parties) : Tour finished, Event completed, Exam passed etc.

All the above can be considered administrative notifications in my opinion.

Most likely we would have an admin setting to enable private discord notifications, nothing more. Then admins would decide to have their own discord application and enable / disable settings there (like having their app id, secret, notifications to be sent as main groups etc)

Reply:
Yes, for a first implementation, that would be sufficient.

In the long term, I think it would be cool to eventually offer each user the ability to choose which notifications they want to receive and where they want to receive them. This way, if someone only wants notifications for rejected PIREPs, they can select that, or if they never check their emails and prefer Discord notifications, they can uncheck the email option, and so on. But that's a long-term goal, not required for the initial version of this feature.

I'll look into how to properly save the `discord_private_channel_id ` tomorrow (sunday) and add it to this PR. The process of sending notifications via this ID should go in another PR I think.

Reply:
Yeah, maybe we can ask a user "where" to get the notifications like email / discord (maybe later more can be added), nothing more 'cause I kindly disagree to allow people to enable / disable administrative notifications as they like.

Then it will be a mess for admins, system will send an administrative notification but the user will not get it because they don't want it ! Sorry, no, you are member and can (and should) not disable everything.

Reply:
Okay in this case we can work on the "where" later